apps/console/src/features/applications/components/edit-application.tsx

```diff
@@ -236,6 +238,19 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
         }
     },[ template ]);
 
+    /**
+     * Check whether the application is an M2M Application
+     */
+    useEffect(() => {
+        if (!template?.id) {    
+            return;
+        }
+
+        if (template?.id == ApplicationTemplateIdTypes.M2M_APPLICATION) {
```

Comment:
```suggestion
        if (template.id == ApplicationTemplateIdTypes.M2M_APPLICATION) {
```

Reply:
Because we are capturing the `undefined` template in the line 245.

Reply:
Agreed. Let's keep the optional chaining operator and remove the first if check.

Reply:
Fixed with 4c374e298f1181dbe40bbbfb9ed9fdae8ef2bb8b